sdk/include/opentelemetry/sdk/metrics/async_instruments.h

```diff
@@ -0,0 +1,176 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+#ifndef ENABLE_METRICS_PREVIEW
+#  include "opentelemetry/metrics/async_instruments.h"
+#  include "opentelemetry/metrics/observer_result.h"
+#  include "opentelemetry/nostd/string_view.h"
+#  include "opentelemetry/sdk/metrics/instruments.h"
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace sdk
+{
+namespace metrics
+{
+
+template <class T>
+class Asynchronous
+{
+public:
+  Asynchronous(nostd::string_view name,
+               const opentelemetry::sdk::instrumentationlibrary::InstrumentationLibrary
+                   *instrumentation_library,
+               MeasurementProcessor *measurement_processor,
+               void (*callback)(opentelemetry::metrics::ObserverResult<T> &),
+               nostd::string_view description = "",
+               nostd::string_view unit        = "")
+      : name_(name),
+        instrumentation_library_{instrumentation_library},
+        measurement_processor_{measurement_processor},
+        callback_(callback),
+        description_(description),
+        unit_(unit)
+  {}
+
+protected:
+  std::string name_;
```

Comment:
Why don't we use `nostd::string_view` for the members?

Reply:
It's better to own these instrument properties. The default ownership of these string_view arguments passed in the constructor is with the instrumentation library, and it may choose to deallocate them anytime.

Reply:
thanks for the clarification.

Reply:
Could the ownership be moved here from the InstrumentationLibrary? Just want to check whether it is possible to avoid the copy here.

Reply:
Yes, it is possible by changing the API for creating metric (with passing `std::string &&` instead of `nostd::string_view`) which would be  ABI breaking change, as the implementation of `std::string` is not ABI compatible.
The meter is created only during SDK configuration and we store these three meter attributes (`name`, `unit`, and `description`) during that time. It is not allocated in the hot path for metric creation and collection.